evap/static/ts/src/notebook.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable  @typescript-eslint/no-non-null-assertion */
+declare const bootstrap: typeof import("bootstrap");
+import { getCookie, setCookie, assertDefinedUnwrap } from "./utils.js";
+import { CSRF_HEADERS } from "./csrf-utils.js";
+
+const NOTEBOOK_COOKIE_NAME = "evap_notebook_open";
+
+if (getCookie("evap_notebook_open") == "true") {
+    new bootstrap.Collapse(assertDefinedUnwrap(document.getElementById("notebook")));
+    new bootstrap.Collapse(assertDefinedUnwrap(document.getElementById("notebookButton")));
+    onShowNotebook();
+}
+
+assertDefinedUnwrap(document.getElementById("notebook-save-button")).addEventListener(
+    "click",
+    function (event: MouseEvent): void {
+        event.preventDefault();
+        const form = assertDefinedUnwrap(document.getElementById("notebook-form")) as HTMLFormElement;
+        const data = new FormData(form);
+
+        const target = assertDefinedUnwrap(event.target) as HTMLButtonElement;
+        const default_label = assertDefinedUnwrap(target.getAttribute("value"));
+        const sending_label = assertDefinedUnwrap(target.getAttribute("data-label-sending"));
+        const error_label = assertDefinedUnwrap(target.getAttribute("value"));
+        target.disabled = true;
+        target.setAttribute("value", sending_label);
+
+        const form_json = JSON.stringify(Object.fromEntries(data.entries()));
+
+        const cooldown_time = 2000;
+        const success_code = 204;
```

Comment:
I think you can inline these.

For `success_code`, you can just test for `response.ok` (https://developer.mozilla.org/en-US/docs/Web/API/Response/ok)